Description:
 Fix vt(4) crash with improper ioctl parameters [EN-15:01] (CVE-2014-0998)
Origin: vendor, https://security.FreeBSD.org/patches/EN-15:01/vt.patch
Bug: https://security.FreeBSD.org/advisories/FreeBSD-EN-15:01.vt.asc
Bug-Debian: https://bugs.debian.org/779194
Applied-Upstream: https://svnweb.freebsd.org/base?view=revision&revision=279264

--- a/sys/dev/vt/vt_core.c
+++ b/sys/dev/vt/vt_core.c
@@ -2340,20 +2340,23 @@
 		}
 		VT_UNLOCK(vd);
 		return (EINVAL);
-	case VT_WAITACTIVE:
+	case VT_WAITACTIVE: {
+		unsigned int idx;
+
 		error = 0;
 
-		i = *(unsigned int *)data;
-		if (i > VT_MAXWINDOWS)
+		idx = *(unsigned int *)data;
+		if (idx > VT_MAXWINDOWS)
 			return (EINVAL);
-		if (i != 0)
-			vw = vd->vd_windows[i - 1];
+		if (idx > 0)
+			vw = vd->vd_windows[idx - 1];
 
 		VT_LOCK(vd);
 		while (vd->vd_curwindow != vw && error == 0)
 			error = cv_wait_sig(&vd->vd_winswitch, &vd->vd_lock);
 		VT_UNLOCK(vd);
 		return (error);
+	}
 	case VT_SETMODE: {    	/* set screen switcher mode */
 		struct vt_mode *mode;
 		struct proc *p1;
